src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/court/CourtEventProcessor.kt

```diff
@@ -142,4 +146,39 @@ class CourtEventProcessor(
   private fun isLibraPerson(libraHearingEvent: LibraHearingEvent) = libraHearingEvent.defendantType == DefendantType.PERSON.value
 }
 
+private fun addCprUUID(
+  message: String,
+  processedDefendants: List<PersonEntity>?,
+): String {
+  val messageParser = JsonPath.parse(message)
+  processedDefendants?.forEach { defendant ->
+    val defendantId = defendant.defendantId
+    val cprUUID = defendant.personKey?.personId.toString()
+    messageParser.put(
+      "$.hearing.prosecutionCases[?(@.defendants[?(@.id == '$defendantId')])].defendants[?(@.id == '$defendantId')]",
```

Comment:
check behaviour if there are no prosecutionCases in the hearing

Reply:
n/a that is the only place where defendants live and the only place court-case-matcher looks